pkg/plugin/v3/scaffolds/internal/templates/config/crd/kustomizeconfig.go

```diff
@@ -35,23 +40,52 @@ func (f *KustomizeConfig) SetTemplateDefaults() error {
 		f.Path = filepath.Join("config", "crd", "kustomizeconfig.yaml")
 	}
 
-	f.TemplateBody = kustomizeConfigTemplate
+	switch f.CRDVersion {
+	case v1beta1:
+		f.TemplateBody = kustomizeConfigTemplatev1beta1
+	default:
+		f.TemplateBody = kustomizeConfigTemplatev1
+	}
 
 	return nil
 }
 
 //nolint:lll
-const kustomizeConfigTemplate = `# This file is for teaching kustomize how to substitute name and namespace reference in CRD
+const kustomizeConfigTemplatev1 = `# This file is for teaching kustomize how to substitute name and namespace reference in CRD
+nameReference:
+- kind: Service
+  version: v1
+  fieldSpecs:
+  - kind: CustomResourceDefinition
+    version: v1
+    group: apiextensions.k8s.io
+    path: spec/conversion/webhook/clientConfig/service/name
+
+namespace:
+- kind: CustomResourceDefinition
+  version: v1
+  group: apiextensions.k8s.io
+  path: spec/conversion/webhook/clientConfig/service/namespace
+  create: false
+
+varReference:
+- path: metadata/annotations
+`
+
+//nolint:lll
+const kustomizeConfigTemplatev1beta1 = `# This file is for teaching kustomize how to substitute name and namespace reference in CRD
 nameReference:
 - kind: Service
   version: v1
   fieldSpecs:
   - kind: CustomResourceDefinition
+    version: v1beta1
     group: apiextensions.k8s.io
     path: spec/conversion/webhookClientConfig/service/name
 
 namespace:
 - kind: CustomResourceDefinition
+  version: v1beta1
   group: apiextensions.k8s.io
   path: spec/conversion/webhookClientConfig/service/namespace
   create: false
```

Comment:
```suggestion
const kustomizeConfigTemplate = `# This file is for teaching kustomize how to substitute name and namespace reference in CRD
nameReference:
- kind: Service
  version: v1
  fieldSpecs:
  - kind: CustomResourceDefinition
    version: {{ .CRDVersion }}
    group: apiextensions.k8s.io
    path: spec/conversion/webhookClientConfig/service/name

namespace:
- kind: CustomResourceDefinition
  version: {{ .CRDVersion }}
  group: apiextensions.k8s.io
  path: spec/conversion/webhookClientConfig/service/namespace
  create: false
```

Reply:
The `path` elements differ between v1 and v1beta1. In general I'd like to have separate templates for different API versions in case we need to further change one version.

Reply:
I would add template conditionals for the path too then. If you use `SetTemplateDefaults` to actually template, changes done after setting the defaults won't be properly reflected in the final scaffolded file, and thus a `v1` config file with `v1beta1` paths may be scaffolded.

Reply:
You're effectively saying if I run

```go
s := KustomizeConfig{}
_ = s.SetTemplateDefaults()
s.WebhookVersion = "v1beta1" // Won't be used in the template
t, _ := template.New("").Parse(s.TemplateBody)
buf := bytes.Buffer{}
_ = t.Execute(&buf, s)
```

Then you may see unexpected behavior? That's a fair reason to use one template. I'll update.

Reply:
Exactly, it will be used only for part of the template.